### PR TITLE
Add coverage for hammer negotiate auth (IDM)

### PR DESCRIPTION
**Pull request by vsedmik**
Introduced changes:
- added test cases for hammer negotiate auth
- added fixture to run the added cases against IDM and AD servers
- moved the IDM enrollment into fixture, added disenroll on IDM teardown to clean up the server side

### CODE REVIEW

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_ldap_authentication.py -k negotiate

File: tests/foreman/destructive/test_ldap_authentication.py [hunk 1]
```diff
@@ -206,39 +218,25 @@ def test_single_sign_on_ldap_ipa_server(
...
-            cmd=f'ipa service-del HTTP/{target_sat.hostname}',
-            hostname=settings.ipa.hostname,
-        )
-        run_command(
-            cmd=f'ipa host-del {target_sat.hostname}',
-            hostname=settings.ipa.hostname,
-        )
+    result = target_sat.execute(f'echo {settings.ipa.password} | kinit {settings.ipa.user}')
+    assert result.status == 0
+    if is_open('BZ:1941997'):
```
Comment (pondrejk) [inline tests/foreman/destructive/test_ldap_authentication.py:226]: do we still need this if bz was closed_errata?

File: tests/foreman/destructive/test_ldap_authentication.py [hunk 2]
```diff
@@ -255,7 +253,8 @@ def test_single_sign_on_ldap_ad_server(
 
     """
     # create the kerberos ticket for authentication
-    target_sat.execute(f'echo {settings.ldap.password} | kinit {settings.ldap.username}')
+    result = target_sat.execute(f'echo {settings.ldap.password} | kinit {settings.ldap.username}')
+    assert result.status == 0
     if is_open('BZ:1941997'):
```
Comment (pondrejk) [inline tests/foreman/destructive/test_ldap_authentication.py:258]: same here

**Review by pondrejk — APPROVED** [review]
ack with minor comments

File: tests/foreman/destructive/test_ldap_authentication.py (hunk 1 above)
Comment (vsedmik) [reply tests/foreman/destructive/test_ldap_authentication.py:226]: I think we could get rid of it and use the variant with slash to catch possible regressions.

File: pytest_fixtures/component/satellite_auth.py [hunk 3]
```diff
@@ -328,35 +331,49 @@ def enroll_idm_and_configure_external_auth(sat):
...
-            f"ipa-client-install --password '{password}' "
-            f'--domain {domain} '
-            f'--server {settings.ipa.hostname} '
-            f'--realm {domain.upper()} -U'
-        )
-        if result.status not in [0, 3]:
-            raise CLIReturnCodeError(result.status, result.stderr, 'Failed to enable ipa client')
+    result = ipa_host.execute(f'ipa host-find {sat.hostname}')
+    if result.status == 0:
+        disenrol_idm(sat)
```
Comment (omkarkhatavkar) [inline pytest_fixtures/component/satellite_auth.py:336]: typo 's/disenrol/disenroll'

File: pytest_fixtures/component/satellite_auth.py [hunk 4]
```diff
@@ -419,70 +434,120 @@ def enroll_ad_and_configure_external_auth(request, ad_data, target_sat):
...
+    if 'IDM' in request.param:
+        enroll_idm_and_configure_external_auth(new_sat)
+        yield new_sat
+        disenrol_idm(new_sat)
+    else:
+        enroll_ad_and_configure_external_auth(request, ad_data, new_sat)
+        yield new_sat
+    new_sat.unregister()
+    new_sat.teardown()
+    Broker(hosts=[new_sat]).checkin()
```
Comment (omkarkhatavkar) [inline pytest_fixtures/component/satellite_auth.py:553]: nice factory!

**Review by omkarkhatavkar — APPROVED** [review]
looks good to me, ACK

**Pull request merged**